### ISSUE TITLE
1

### ISSUE THREAD

**Issue by ginesufa** [ginesufa/calc#1]
<!DOCTYPE html>
<html lang="ru">
<head>
	<meta charset="UTF-8">
	<title></title>
</head>

<body>
	<div id="form-calc">
		<div class="calc_group-input">
			<label for=""><input type="radio" name="test1" value="0">0</label>
			<label for=""><input type="radio" name="test1" value="105">105</label>
			<label for=""><input type="radio" name="test1" value="250">250
			<p><span class="calc_group-input"><label><input type="checkbox" name="test5" value="2000"> Капот</label></span> <span class="calc_group-input"><label><input type="checkbox" name="test5" value="8000"> Арки</label></span> <span class="calc_group-input"><label><input type="checkbox" name="test5" value="10000"> Двери</label></span></p>
			</label>
		</div>
		<div id="summ">0</div>		
	</div>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
<script>
	var calc = {
		summ: 0, // сумма изначально 0
		valueArray: (function () { //массив изначально создается на основе данных value выбранных кнопок
			var array = [],
				arrayLength = $("#form-calc .calc_group-input").length;
			for (var i = 0; i < arrayLength; i++) {
				array[i] = parseInt($("#form-calc .calc_group-input").eq(i).find("input:checked").attr("value")) || 0;
			};
			return array;
		})(),
		summation: function () { //суммирует значения массива с данными
			var summ = 0,
				i = this.valueArray.length - 1;
			for (; i >= 0; i--) {
				summ += this.valueArray[i];
			};
			this.summ = summ;
			$("#summ").html(calc.summ);
		},
		changeEvent: function () {	//подключение обработчика событий
			$("input[type='radio']").change(function () {	//для радиокнопок
				var element = event.target,
					elementValue = parseInt(element.value),
					elementId = $(element).parents(".calc_group-input").index();
				calc.valueArray[elementId] = elementValue;
				calc.summation ();
			});
			$("input[type='checkbox']").change(function () {	//для чекбоксов
				var element = event.target,
					elementValue = parseInt(element.value),
					elementId = $(element).parents(".calc_group-input").index();
				if (!element.checked) {
					elementValue = 0;
				};
				calc.valueArray[elementId] = elementValue;
				calc.summation ();
			});
		},
		init: function () {
			calc.summation ();
			calc.changeEvent ();
		}
	};
	calc.init ();
</script>

</body>
</html>
